src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/validate/PNCIdentifier.kt

```diff
@@ -5,12 +5,12 @@ import java.time.LocalDate
 
 const val LONG_PNC_ID_LENGTH = 10
 
-data class PNCIdentifier(private val inputPncId: String? = null) {
-  val inputPnc: String?
-    get() = inputPncId
+class PNCIdentifier(private val inputPncId: String? = null) {
+  private val storedPncId: String? = inputPncId?.uppercase()
+  // always use canonical format when storing and comparing
 
   val pncId: String?
-    get() = toCanonicalForm(inputPncId)
+    get() = toCanonicalForm(storedPncId)
```

Comment:
I wonder if we might be better with the factory pattern to create a PNC and do all the necessary logging?